commit fc13a78e1f68fa5ca41280a397a046a6eff6cc9b
Merge: 06961fbbbd7a b688f369ae0d
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Mon Mar 24 15:18:08 2025 -0700

    Merge tag 'hardening-v6.15-rc1' of git://git.kernel.org/pub/scm/linux/kernel/git/kees/linux
    
    Pull hardening updates from Kees Cook:
     "As usual, it's scattered changes all over. Patches touching things
      outside of our traditional areas in the tree have been Acked by
      maintainers or were trivial changes:
    
       - loadpin: remove unsupported MODULE_COMPRESS_NONE (Arulpandiyan
         Vadivel)
    
       - samples/check-exec: Fix script name (Mickaël Salaün)
    
       - yama: remove needless locking in yama_task_prctl() (Oleg Nesterov)
    
       - lib/string_choices: Sort by function name (R Sundar)
    
       - hardening: Allow default HARDENED_USERCOPY to be set at compile
         time (Mel Gorman)
    
       - uaccess: Split out compile-time checks into ucopysize.h
    
       - kbuild: clang: Support building UM with SUBARCH=i386
    
       - x86: Enable i386 FORTIFY_SOURCE on Clang 16+
    
       - ubsan/overflow: Rework integer overflow sanitizer option
    
       - Add missing __nonstring annotations for callers of
         memtostr*()/strtomem*()
    
       - Add __must_be_noncstr() and have memtostr*()/strtomem*() check for
         it
    
       - Introduce __nonstring_array for silencing future GCC 15 warnings"
    
    * tag 'hardening-v6.15-rc1' of git://git.kernel.org/pub/scm/linux/kernel/git/kees/linux: (26 commits)
      compiler_types: Introduce __nonstring_array
      hardening: Enable i386 FORTIFY_SOURCE on Clang 16+
      x86/build: Remove -ffreestanding on i386 with GCC
      ubsan/overflow: Enable ignorelist parsing and add type filter
      ubsan/overflow: Enable pattern exclusions
      ubsan/overflow: Rework integer overflow sanitizer option to turn on everything
      samples/check-exec: Fix script name
      yama: don't abuse rcu_read_lock/get_task_struct in yama_task_prctl()
      kbuild: clang: Support building UM with SUBARCH=i386
      loadpin: remove MODULE_COMPRESS_NONE as it is no longer supported
      lib/string_choices: Rearrange functions in sorted order
      string.h: Validate memtostr*()/strtomem*() arguments more carefully
      compiler.h: Introduce __must_be_noncstr()
      nilfs2: Mark on-disk strings as nonstring
      uapi: stddef.h: Introduce __kernel_nonstring
      x86/tdx: Mark message.bytes as nonstring
      string: kunit: Mark nonstring test strings as __nonstring
      scsi: qla2xxx: Mark device strings as nonstring
      scsi: mpt3sas: Mark device strings as nonstring
      scsi: mpi3mr: Mark device strings as nonstring
      ...

diff --cc MAINTAINERS
index 5da1d8086101,a1900962ced9..b04862031c91
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@@ -12591,8 -12586,9 +12591,9 @@@ F:	Documentation/ABI/testing/sysfs-kern
  F:	arch/*/configs/hardening.config
  F:	include/linux/overflow.h
  F:	include/linux/randomize_kstack.h
+ F:	include/linux/ucopysize.h
  F:	kernel/configs/hardening.config
 -F:	lib/usercopy_kunit.c
 +F:	lib/tests/usercopy_kunit.c
  F:	mm/usercopy.c
  F:	security/Kconfig.hardening
  K:	\b(add|choose)_random_kstack_offset\b